wallet-new/src/Cardano/Wallet/API/V1/LegacyHandlers/Info.hs

```diff
@@ -52,3 +58,9 @@ getInfo Diffusion{..} ntpStatus = do
             { timeDifferenceFromNtpServer = fmap V1.mkLocalTimeDifference timeDifference
             }
         }
+  where
+    checkNtpBlocking = do
+        s <- readTVar ntpStatus
+        case s of
+            NtpSyncPending -> checkNtpBlocking
+            _              -> newTVar s
```

Comment:
Doesn't this use a lot of CPU cycles :/ ? Does it make sense to introduce a small delay between each call to `checkNtpBlocking`?

Reply:
Hm, yes you are right. I got confused by something Marcin told me (and thought  readTVar will block - which is obviously wrong!). Will fix

Reply:
Aha - Marcin explained me. There is a `retry` in STM which will block is value in `TVar` didn't change.

thanks @coot